src/Sarif.Multitool/FileWorkItemsCommand.cs

```diff
@@ -2,15 +2,105 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Globalization;
+using System.Linq;
 
 namespace Microsoft.CodeAnalysis.Sarif.Multitool
 {
     public class FileWorkItemsCommand : CommandBase
     {
-        public int Run(FileWorkItemsOptions fileWorkItemsOptions)
+        private static readonly string[] s_knownFilingTargetTypes = new[] { "github", "azuredevops" };
+        private static readonly string[] s_knownFilteringStrategies = new[] { "new", "all" };
+        private static readonly string[] s_knownGroupingStrategies = new[] { "perresult" };
+
+        private FileWorkItemsOptions _options;
+
+        public int Run(FileWorkItemsOptions options)
```

Comment:
>public int Run(FileWorkItemsOptions options) [](start = 8, length = 44)

ryley says: If options are required for the functionality, shouldn't the validation (and setting of _options) happen in a constructor?

Reply:
This is the Commandline OSS component pattern. By convention, it creates and throws away a single instance. The design Ryley mentions could work, except that a ctor, obviously, cannot return an exit code (and is dedicated to initialization only). So to implement his suggestion, commandline lib would need to find and invoke the ctor, then find and invoke the Run method. I think they thought it more straightforward to simply use a single method. Also, it is a fine design to create an object that provides a guarantee of allowing for coherent initialization/operation against a method like this. So the design can be justified on its own terms, imo.

---
In reply to: [306578717](https://github.com/microsoft/sarif-sdk/pull/1583#discussion_r306578717) [](ancestors = 306578717)